docs/guide/building-an-advanced-scene.md

```diff
@@ -4,238 +4,456 @@ type: guide
+}}` variable in the template, which we can pass using the data attributes:
+
+```html
+<a-assets>
+  <!-- ... -->
+  <!-- Specify Nunjucks. -->
+  <script id="link" type="text/nunjucks">
+    <a-plane class="link" height="1" width="1"
+             material="shader: flat; src: {{ thumb }}"
+             sound="on: cursor-click; src: audio/click.ogg"></a-plane>
+  </script>
+</a-assets>
+
+<!-- ... -->
+
+<!-- Pass image sources to the template. -->
+<a-entity template="src: #plane" data-thumb="#city-thumb></a-entity>
+<a-entity template="src: #plane" data-thumb="#cubes-thumb"></a-entity>
+<a-entity template="src: #plane" data-thumb="#sechelt-thumb"></a-entity>
+```
+
+Here, the template component allows us to not have to repeat code and keep our
+scene clean.
+
+### Layout Component
+
+But the position of the links are all the same so they will overlap. While we
+could manually position each link, we could use the [layout component][layout]
+to do it for us. The layout component will automatically position its children
+to the specified layout.
+
+We create a wrapper entity around our links and attach the layout component
+using the `line` layout:
+
+```html
+<a-entity id="links" layout="layout: line; margin: .75" position="-3 -1 -4">
+  <a-entity template="src: #plane" data-thumb="#city-thumb></a-entity>
+  <a-entity template="src: #plane" data-thumb="#cubes-thumb"></a-entity>
+  <a-entity template="src: #plane" data-thumb="#sechelt-thumb"></a-entity>
+</a-entity>
+```
+
+Now our links are no longer overlapping without us having to calculate and
+fiddle with positions.
+
+### Event Set Component
+
+Lastly, we'll add some visual feedback to our links. We want them to scale up
+and scale back when they are hovered or clicked. This involves writing an event
+listener to do `setAttribute`s on the [scale component][scale] in response to
+[cursor events][cursor-events]. This is a fairly common pattern so there is a
+[event set component][event-set] that does `setAttribute` in response to
+events.
+
+The event set component defines a unique syntax to be able to define multiple
+event handlers. Wheres single-property component data represents a value, and
+multi-property component data represents an object, the event set component is
+comma-separated and represents an array of objects.
+
+Let's attach event setters on our links to scale them up when they are gazed
+over, scale them down as they are being clicked, and scale them back when they
+are no longer gazed upon. We are mimicking CSS hover states from the 2D web.
+We can specify event names with `_event` properties, and the rest of the
+properties define the `setAttibute` calls:
+
+```html
+<a-assets>
+  <!-- ... -->
+  <script id="link" type="text/nunjucks">
+    <a-plane class="link" height="1" width="1"
+             material="shader: flat; src: {{ thumb }}"
+             sound="on: cursor-click; src: audio/click.ogg"
```

Comment:
Sound doesn't support selectors yet and yeah it doesn't have a URL wrapper. There's an issue, probably should get that wrapped up.